### ISSUE TITLE
Why does the cpu bias_op support only up to five dims?

### ISSUE THREAD

**Issue by idofr** [tensorflow/tensorflow#35308]
I'm currently using tf.1.14 on Windows 10, but it doesn't seem to matter too much.

I have a high-dimensional tensor object which I try to process with a Keras Dense layer. On the GPU everything works perfectly fine, but if I want to run the network on the CPU I get the following error "Only ranks up to 5 supported:.... ".  A quick look into the C code shows that the dimensionality handling is, indeed, hardcoded:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/bias_op.cc#L102

My question is only why? It seems like it might have something to do with the way `GetBiasValueDims()` is written, but not necessarily.

**Comment by ezhulenev** [comment]
This was done originally to reduce the number of instantiated templates, however this could be always computed with tensors:

(1) Rank 2 for NHWC -> fuse all dimensions before last
(2) Rank 3 for NCHW -> fuse all dimensions after dim 1

I'm not sure when I'll have cycles to add this fix myself, but it should be rather trivial. Would gladly review external contribution.

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this, as you are using an older version of tensorflow which is officially considered end of life . We recommend that you upgrade to the latest 2.x version and let us know if the issue still persists in newer versions. Please open a new issue for any help you need against 2.x, and we will get you the right help. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, please provide us with more information.
